src/core/weightedGraph2.js

```diff
@@ -37,89 +65,252 @@ export type WeightedEdge = {|
  * WeightedGraph2 should completely replace it.
  */
 export class WeightedGraph2 {
```

Comment:
It would be nice to have some interface that describes what `WeightedGraph2` does, similar to `WeightsI` or `NodeWeightsI`.

Reply:
My intuition was only to spec out an interface if it is meant to be utilized by other classes, but I like this idea,

Reply:
I find it helpful to understand the code on a high level quickly, even if it's just for that